### PR TITLE
fix(csi-578): updated xstate version

### PR DESCRIPTION
**Pull request by geka-evk**
- Made browser-based _xstate/inspect_ possibility optional to avoid memory leak (set `STATE_MACHINE_INSPECT_ENABLED` env var to _true_ to enable introspection)
- Updated TS config to avoid `/src` folder inside `/dist` folder
- Removed some _devDependencies_ from production build, updated xstate version to avoid possible [issue](https://github.com/statelyai/xstate/issues/3616)
- Updated _docker-compose.yml_ to be able to run `men-api` locally
- Added metrics (Prometheus/Grafana)
- Updated Node.js version to 18 in Dockerfile

### CODE REVIEW

File: docker/keycloak/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+COPY --from=builder /opt/keycloak/ /opt/keycloak/
+
+# change these values to point to a running postgres instance
+ENV KC_DB=mysql
+#ENV KC_DB_URL=<DBURL>
+#ENV KC_DB_USERNAME=<DBUSERNAME>
+#ENV KC_DB_PASSWORD=<DBPASSWORD>
+#ENV KC_HOSTNAME=localhost
+
+ENTRYPOINT ["/opt/keycloak/bin/kc.sh"]
```
Comment (vijayg10) [inline docker/keycloak/Dockerfile:25]: Why are we building our own keycloak?

File: src/config.ts [hunk 2]
```diff
@@ -12,8 +12,6 @@ import fs from 'fs';
 import { from } from 'env-var';
 import yaml from 'js-yaml';
 
-require('dotenv').config();
```
Comment (vijayg10) [inline src/config.ts:15]: Does it work if we remove dotenv here?

File: package.json [hunk 3]
```diff
@@ -17,13 +17,11 @@
...
   },
   "scripts": {
-    "clean": "rimraf dist/",
-    "build": "npm run clean && tsc && npm run copy-files && rm dist/src/lib/cacheDatabase/migrations/*.ts",
-    "start": "npm run start:no-build",
-    "start:no-build": "node -r ./tsconfig-paths.js dist/src/index.js",
-    "start:build": "npm run build && npm run start:no-build",
-    "start:dev": "ts-node -r dotenv/config -r tsconfig-paths/register -P ./tsconfig.json src/index.ts",
-    "copy-files": "copyfiles -u 1 src/**/*.yaml dist/src",
+    "build": "tsup && npm run copy-files",
```
Comment (vijayg10) [inline package.json:20]: Are we going to use this new "tsup" command to build? If yes, you may need to present this in our group about its advantages and the reason to migrate.

File: docker-compose.yaml [hunk 4]
```diff
@@ -1,5 +1,30 @@
-version: '3'
+version: "3.9"
+
+volumes:
+  mysql-data:
+  prom-data:
+  grafana-data:
+
 services:
+  men-api:
```
Comment (vijayg10) [inline docker-compose.yaml:9]: What is men-api?

File: test/unit/model/stateMachine/connectorConfig.test.ts [hunk 5]
```diff
@@ -8,8 +8,6 @@
  *       Yevhen Kyriukha <yevhen.kyriukha@modusbox.com>                   *
  **************************************************************************/
 
-import 'tsconfig-paths/register';
-
 import { ConnectorConfig } from '@app/lib/model/stateMachine/states';
```
Comment (vijayg10) [inline test/unit/model/stateMachine/connectorConfig.test.ts:11]: ts paths support is removed how is this import statement working?

File: package.json [hunk 6]
```diff
@@ -69,23 +66,22 @@
     "oas-validator": "^5.0.0",
     "openapi-backend": "^5.9.1",
     "redis": "^4.1.0",
-    "reflect-metadata": "^0.1.13",
-    "ts-node": "^10.8.0",
-    "tsconfig-paths": "^4.0.0",
```
Comment (vijayg10) [inline package.json:74]: Don't we need ts path support?

File: docker/keycloak/Dockerfile (hunk 1 above)
Comment (geka-evk) [reply docker/keycloak/Dockerfile:25]: It's just a way how I spun `keycloak` in my previous tasks. If we have a reference, how we should use `keycloak`  locally, I'll update this approach

File: src/config.ts (hunk 2 above)
Comment (geka-evk) [reply src/config.ts:15]: We don't need to use `dotenv` in production env, we use k8s ConfigMaps/Secrets for setting all needed env vars. So I should have this line in code.  To use .env files locally we can utilize a better approach:
https://github.com/pm4ml/mojaloop-payment-manager-management-api/pull/32/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R23

And yes - it works on k8s env:
image

File: docker-compose.yaml (hunk 4 above)
Comment (geka-evk) [reply docker-compose.yaml:9]: oh, it should be m`a`n-api, stands from management-api )

File: test/unit/model/stateMachine/connectorConfig.test.ts (hunk 5 above)
Comment (geka-evk) [reply test/unit/model/stateMachine/connectorConfig.test.ts:11]: it works coz we have this setting in tsconfig: https://github.com/pm4ml/mojaloop-payment-manager-management-api/pull/32/files#diff-b55cdbef4907b7045f32cc5360d48d262cca5f94062e353089f189f4460039e0R37

File: package.json (hunk 6 above)
Comment (geka-evk) [reply package.json:74]: I don't see any reason why it's needed

File: package.json (hunk 3 above)
Comment (geka-evk) [reply package.json:20]: It's just a bit more performant way to transpile TS to JS with some extra features like: generate JS for multiple module targets (e.g. CJS, ESM), bundling.  It allowed us inside _inter-scheme-proxy-cache-lib_ to generate different JS code to use it with any modules with ease:
image

https://blog.atomrc.dev/p/typescript-library-compilation-2023/

Comment (vijayg10) [reply package.json:20]: No we don't have strict rule, but we defined a template repo for typescript projects with all the tools we want to use in Infitx. If there are any deviations in the major components like this, we need to discuss. You just need to discuss and convince the team, we are more than happy to try new tech.

File: docker/keycloak/Dockerfile (hunk 1 above)
Comment (vijayg10) [reply docker/keycloak/Dockerfile:25]: Ok.
These docker compose files have keycloak deployments
https://github.com/pm4ml/on-premise-deploy/blob/master/docker-compose/docker-compose.yaml
https://github.com/mojaloop/reporting-hub-bop-shell/blob/master/docker-compose.yaml
Lets try to use the official docker image if possible

File: src/config.ts (hunk 2 above)
Comment (vijayg10) [reply src/config.ts:15]: Oh ok. If it works without importing, then that's ok. Infact that is my question, got it.

File: package.json (hunk 6 above)
Comment (vijayg10) [reply package.json:74]: Ok, that's surprising. Without having that in dependencies, we can still import? Anyway if you are sure, lets resolve this comment.

File: test/unit/model/stateMachine/connectorConfig.test.ts (hunk 5 above)
Comment (vijayg10) [reply test/unit/model/stateMachine/connectorConfig.test.ts:11]: Ok, that's surprising. Without having that in dependencies, we can still import? Anyway if you are sure, lets resolve this comment.

File: docker/keycloak/Dockerfile (hunk 1 above)
Comment (geka-evk) [reply docker/keycloak/Dockerfile]: Updated

**Review by vijayg10 — APPROVED** [review]

**Pull request merged**